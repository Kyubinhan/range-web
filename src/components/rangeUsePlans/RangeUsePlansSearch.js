import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon, Input } from 'semantic-ui-react';

const propTypes = {
  handleSearchInput: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
}

class RangeUsePlansSearch extends Component {
  state = {
    search: '',
  }
  
  handleInput = (e) => {
    const { id, value } = e.target;
    
    this.setState({
      [id]: value
    }, () => {
      this.props.handleSearchInput(value);
    });
  }
  
  render() {
    const { search } = this.state;
    const { placeholder } = this.props;

    return (
      <div className="range-use-plans-search">
        <Input
          fluid
          icon
          loading={false} 
          placeholder={placeholder}>
          <input 
            id="search"
            value={search}
            onChange={this.handleInput}
          />
          <Icon name="search" />
        </Input>
      </div>
    );
  }
}
RangeUsePlansSearch.propTypes = propTypes;
export default RangeUsePlansSearch;