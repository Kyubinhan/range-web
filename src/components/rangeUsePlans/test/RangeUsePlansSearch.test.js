import React from 'react';
import { shallow } from 'enzyme';
import RangeUsePlansSearch from '../RangeUsePlansSearch';

const props = {};
const setupProps = () => {
  props.handleSearchInput = jest.fn();
  props.placeholder = "placeholder";
  props.label = "label";
};

beforeEach(() => {
  setupProps();
});

describe('RangeUsePlansSearch', () => {
  xit('renders correctly', () => {
    const wrapper = shallow(<RangeUsePlansSearch {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('set `search` state when typing terms for searching', () => {
    const wrapper = shallow(<RangeUsePlansSearch {...props} />);
    const mockTerm = "hello";

    wrapper.find('input#search').simulate('change', { target: { id: 'search', value: mockTerm }});
    expect(wrapper.state().search).toEqual(mockTerm);
    expect(props.handleSearchInput).toHaveBeenCalledWith(mockTerm);
  });

  describe('Event handlers', () => {
    it('`handleInput` set state and calls handleSearchInput', () => {
      const wrapper = shallow(<RangeUsePlansSearch {...props} />);
      const mockId = 'id';
      const mockValue = 'value';

      wrapper.instance().handleInput({ target: { id: mockId, value: mockValue }});
      expect(wrapper.state()[mockId]).toEqual(mockValue);
      expect(props.handleSearchInput).toHaveBeenCalledWith(mockValue);
    });
  });
});