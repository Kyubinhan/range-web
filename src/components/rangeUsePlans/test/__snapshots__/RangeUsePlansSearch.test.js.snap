// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeUsePlansSearch renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RangeUsePlansSearch
    handleSearchInput={[Function]}
    label="label"
    placeholder="placeholder"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Input
        fluid={true}
        icon={true}
        loading={false}
        placeholder="placeholder"
        type="text"
>
        <input
                id="search"
                onChange={[Function]}
                value=""
        />
        <Icon
                as="i"
                name="search"
        />
</Input>,
      "className": "range-use-plans-search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <input
            id="search"
            onChange={[Function]}
            value=""
/>,
          <Icon
            as="i"
            name="search"
/>,
        ],
        "fluid": true,
        "icon": true,
        "loading": false,
        "placeholder": "placeholder",
        "type": "text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "search",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "name": "search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Input
          fluid={true}
          icon={true}
          loading={false}
          placeholder="placeholder"
          type="text"
>
          <input
                    id="search"
                    onChange={[Function]}
                    value=""
          />
          <Icon
                    as="i"
                    name="search"
          />
</Input>,
        "className": "range-use-plans-search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <input
              id="search"
              onChange={[Function]}
              value=""
/>,
            <Icon
              as="i"
              name="search"
/>,
          ],
          "fluid": true,
          "icon": true,
          "loading": false,
          "placeholder": "placeholder",
          "type": "text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "search",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "name": "search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
